### PR TITLE
Fixes #26065: OpenSCAP report needs to be rendered without iframe with sanitized html

### PR DESCRIPTION
**Pull request by clarktsiory**
https://issues.rudder.io/issues/26065

We have to read the report file (as the API does) in the snippet, as pass the information down, to create buttons with a download action (similar to https://github.com/Normation/rudder/pull/5941).

The iframe can be used to render the original report by putting the sanitized content as string in [the srcdoc attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#srcdoc)

Result : 
Screenshot from 2024-12-13 16-16-40

### CODE REVIEW

**Comment by clarktsiory** [comment]
Commit modified

**Comment by clarktsiory** [comment]
Commit modified

**Comment by clarktsiory** [comment]
Commit modified

File: openscap/src/main/scala/com/normation/plugins/openscappolicies/extension/OpenScapNodeDetailsExtension.scala [hunk 1]
```diff
@@ -129,13 +145,46 @@ class OpenScapNodeDetailsExtension(
...
-    ".original [download]" #> s"OpenSCAP report for ${hostname} (${nodeId.value}).html"
+  private def downloadReportButton(fileName: String, bytes: Array[Byte], sanitized: Boolean) = {
+    val base64   = Base64.getEncoder().encodeToString(bytes)
+    val download = Run(
+      s"""saveByteArray("${fileName}", "application/zip", base64ToArrayBuffer("${base64}"));"""
+    )
+
+    val btn = if (sanitized) {
+      SHtml.ajaxButton(
+        <span><i class="fa fa-download text-light"></i>Download report</span>,
```
Comment (fanf) [inline openscap/src/main/scala/com/normation/plugins/openscappolicies/extension/OpenScapNodeDetailsExtension.scala:175]: We don't want a download here, we want to open a tab with the report. That allows to see it in full page, with link navigation available, which is mandatory to be able to see anything.

**Review by fanf — CHANGES_REQUESTED** [review]

**Comment by clarktsiory** [comment]
PR updated with a new commit

**Review by fanf — APPROVED** [review]

**Pull request merged**